teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -106,34 +127,72 @@ const CatalogList: React.FC = () => {
         announceToScreenReader(lf("Added '{0}' to checklist.", getReadableCriteriaTemplate(c)));
     }
 
+    function getItemIdForTag(tag: string) {
+        return `accordion-item-${tag}`;
+    }
+
+    function onTagExpandToggled(tag: string, expanded: boolean) {
```

Comment:
this being async seems dangerous. looking at the implementation below of these two functions, it seems like hammering on this button could cause some issues with multiple copies of the same tag being added to the state

Reply:
Ah true...I'll just remove the async stuff. Most of our local storage stuff isn't async anyway.